### PR TITLE
[tests-only][full-ci]refactoring the tag creating step using data table

### PR DESCRIPTION
**Pull request by nabim777**
## Description
This PR consists of refactoring `created tags` step where multiple tags are created for a file.
- new step is added where the multiple tags can be added with multiple files

## Related Issues
 https://github.com/owncloud/ocis/issues/6606#event-9635806756

## Motivation and Context
several tags may only be added to one file at a time in the previous step, thus that step must be repeated in order to add tags to several files. That will lengthen the scenario.
So, a new step has been added in which several tags can be placed in multiple files at once and it will cut short the length of the scenario

## How Has This Been Tested?

- test environment:
- locally
- ci

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests only (no source changes)

## Checklist:

- [X] Code changes
- [ ] Unit tests added
- [X] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 1]
```diff
@@ -18,12 +18,11 @@ Feature: full text search
       | fileInRootLevel.txt                              |
       | folderWithFile/fileInsideFolder.txt              |
       | folderWithFile/subFolder/fileInsideSubFolder.txt |
-    And user "Alice" has created the following tags for file "fileInRootLevel.txt" of the space "Personal":
-      | tag1 |
-    And user "Alice" has created the following tags for file "folderWithFile/fileInsideFolder.txt" of the space "Personal":
-      | tag1 |
-    And user "Alice" has created the following tags for file "folderWithFile/subFolder/fileInsideSubFolder.txt" of the space "Personal":
-      | tag1 |
+    And user "Alice" has created the following tags for files of the space "Personal":
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/fullSearch.feature:21]:
```suggestion
    And user "Alice" has tagged the following files of the space "Personal":
```

File: tests/acceptance/features/bootstrap/TagContext.php [hunk 2]
```diff
@@ -104,6 +103,26 @@ public function theUserHasCreatedFollowingTags(string $user, string $fileOrFolde
 		$this->featureContext->theHttpStatusCodeShouldBe(200);
 	}
 
+	/**
+	 * @Given /^user "([^"]*)" has created the following tags for (folders|files)\s? of the space "([^"]*)":$/
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/TagContext.php:107]:
```suggestion
	 * @Given /^user "([^"]*)" has created the following tags for (folders|files) of the space "([^"]*)":$/
```

File: tests/acceptance/features/bootstrap/TagContext.php [hunk 3]
```diff
@@ -104,6 +103,26 @@ public function theUserHasCreatedFollowingTags(string $user, string $fileOrFolde
...
+	 * @param string $user
+	 * @param string $filesOrFolders (files|folders)
+	 * @param string $space
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userHasCreatedTheFollowingTagsForFilesOfTheSpace(string $user, string $filesOrFolders, string $space, TableNode $table):void {
+		$rows = $table->getHash();
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/TagContext.php:118]: check that `path` and `tagName` columns are in the data table
```suggestion
	public function userHasCreatedTheFollowingTagsForFilesOfTheSpace(string $user, string $filesOrFolders, string $space, TableNode $table):void {
		$this->featureContext->verifyTableNodeColumns($table, ["path", "tagName")
		$rows = $table->getHash();
```

File: tests/acceptance/features/bootstrap/TagContext.php [hunk 4]
```diff
@@ -104,6 +103,27 @@ public function theUserHasCreatedFollowingTags(string $user, string $fileOrFolde
...
+	 * @param string $space
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userHasCreatedTheFollowingTagsForFilesOfTheSpace(string $user, string $filesOrFolders, string $space, TableNode $table):void {
+        $this->featureContext->verifyTableNodeColumns($table, ["path", "tagName"]);
+        $rows = $table->getHash();
+		foreach ($rows as $row) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/TagContext.php:120]:
```suggestion
	public function userHasCreatedTheFollowingTagsForFilesOfTheSpace(string $user, string $filesOrFolders, string $space, TableNode $table):void {
                $this->featureContext->verifyTableNodeColumns($table, ["path", "tagName"]);
                $rows = $table->getHash();
		foreach ($rows as $row) {
```

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by amrita-shrestha — APPROVED** [review]
lgtm

**Pull request merged**